### PR TITLE
Initial commit of file storage service

### PR DESCRIPTION
**Pull request by dav-defra**

### CODE REVIEW

File: plants-file-storage-service/applicationinsights-agent-3.0.jar [hunk 1]
```diff
@@ -0,0 +1 @@
+Not Found
```
Comment (nadeem-iqbal) [inline plants-file-storage-service/applicationinsights-agent-3.0.jar]: Do these jar files needs to be checked in?

File: plants-file-storage-service/dependency-reduced-pom.xml [hunk 2]
```diff
@@ -0,0 +1,165 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nadeem-iqbal) [inline plants-file-storage-service/dependency-reduced-pom.xml]: Do we need this reduced pom? We don't seems to have it in the other repo.

File: plants-file-storage-service/src/test/java/uk/gov/defra/plants/filestorage/service/TemplateUploadServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+import uk.gov.defra.plants.filestorage.provider.StreamingOutputProvider;
+import uk.gov.defra.plants.filestorage.representation.DocumentCategory;
+import uk.gov.defra.plants.filestorage.resource.TemplateResource;
+import uk.gov.defra.plants.filestorage.service.sanitise.SanitiseService;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TemplateUploadServiceTest {
+
+  private static final String ORIGINAL_FILE_NAME = "original_file_name.pdf";
+  private static final String ADMIN_ID = "cd235a5b-de4b-40bf-98e7-8b4bd330ffe4";
```
Comment (nadeem-iqbal) [inline plants-file-storage-service/src/test/java/uk/gov/defra/plants/filestorage/service/TemplateUploadServiceTest.java:83]: Can we please double check these ID's are only for test purpose and are safe to use in these tests?

File: plants-file-storage-service/applicationinsights-agent-3.0.jar (hunk 1 above)
Comment (dav-defra) [reply plants-file-storage-service/applicationinsights-agent-3.0.jar]: removed

File: plants-file-storage-service/src/test/java/uk/gov/defra/plants/filestorage/service/TemplateUploadServiceTest.java (hunk 3 above)
Comment (dav-defra) [reply plants-file-storage-service/src/test/java/uk/gov/defra/plants/filestorage/service/TemplateUploadServiceTest.java:83]: This is a test id only from demo data

**Review by nadeem-iqbal — APPROVED** [review]

**Review by imtiaz-defra — APPROVED** [review]

**Pull request merged**